Sources/Parsing/Parsers/Map.swift

```diff
@@ -25,18 +25,18 @@ extension Parsers {
     public let upstream: Upstream
 
     /// The closure that transforms output from the upstream parser.
-    public let transform: (Upstream.Output) -> NewOutput
+    public let transform: (Upstream.Output) throws -> NewOutput
 
     @inlinable
-    public init(upstream: Upstream, transform: @escaping (Upstream.Output) -> NewOutput) {
+    public init(upstream: Upstream, transform: @escaping (Upstream.Output) throws -> NewOutput) {
       self.upstream = upstream
       self.transform = transform
     }
 
     @inlinable
     @inline(__always)
-    public func parse(_ input: inout Upstream.Input) rethrows -> NewOutput {
-      self.transform(try self.upstream.parse(&input))
+    public func parse(_ input: inout Upstream.Input) throws -> NewOutput {
+      try self.transform(self.upstream.parse(&input))
```

Comment:
Losing `rethrows` is probably the main reason why we'd want a dedicated operator for a throwing `map`. Combine does the same thing with `tryMap`. We're definitely open to adding `tryMap`! Want to start a discussion about `tryMap` and other potential throwing operators?

Reply:
I've started one at #121.